dashboard.py

```
@@ -191,6 +206,51 @@ def get_sales_totals(app_id):
     return row
 
 
+def get_last_fetched_date(table, app_id):
+    conn = get_conn()
+    row = conn.execute(f"SELECT MAX(date) FROM {table} WHERE app_id=?", (str(app_id),)).fetchone()
```

Ruff (0.15.7)

[error] Possible SQL injection vector through string-based query construction

(S608)